src/box/memtx_tx.c

```diff
@@ -840,8 +849,6 @@ memtx_tx_story_new(struct space *space, struct tuple *tuple,
 	struct memtx_tx_stats *stats = &txm.story_stats[story->status];
 	memtx_tx_stats_collect(stats, pool->objsize);
 	story->tuple_is_retained = false;
-	if (!tuple_is_referenced_to_pk)
-		memtx_tx_story_track_retained_tuple(story);
```

Comment:
Oh, that's because the previous logic was quite confusing.
Look, previously this `memtx_tx_story_track_retained_tuple` was called if `tuple_is_referenced_to_pk == false`. You can see that `false` was used only in one case - for added story in `memtx_tx_history_add_insert_stmt`. On the other hand later in `memtx_tx_history_add_insert_stmt` this added story was passed to `memtx_tx_ref_to_primary`, where it was immediately untracked as retained tuple.. So in this patch just does not do this meaningless work.